### DISCUSSION TITLE
Azure AD as OAuth provider - single tenant not supported?

### DISCUSSION THREAD

**Discussion by solvemon-vh** [supabase/supabase#1071] [Questions]
Hi

I'm experimenting with an application for internal use at my company, where only users from our existing AD should have access. For this I am using Azure as an OAuth provider, and have set the "Support account types" to "Single Tenant".

When performing the sign-in action I am sent to the MS sign-in dialog, I sign in, and afterwards am redirected back to my app.

However I assume I have made a mistake somewhere, as the URL parameters includes the following error:

```Application 'xxxxx' is not configured as a multi-tenant application. Usage of the /common endpoint is not supported for such applications created after '10/15/2018'. Use a tenant-specific endpoint or configure the application to be multi-tenant.```

As mentioned I'm not interested in allowing users outside of my organisation to authenticate, so is there some configuration that needs to be done for this to work?

Thanks in advance for any help!

**Comment by dilly-sixmo** [comment]
I'm wondering the same thing

**Comment by hamishakl** [comment]
Did you end up finding a solution?

**Comment by solvemon** [comment]
I have not worked on this project for about 2 years now, so I don't know if this still is a limitation with Supabase Azure AD integration.

My solution ended up with a manual user-set up solution, where an application admin would go in and pre-register allowed users. Other users, who would successfully sign in, would be immediately signed out again, with a message to contact our app admins for sign-up.

Absolutely not a very elegant solution, but for a few hundred employees in the company, not a very big problem.

**Comment by Bigood** [comment]
**As far as I remember, the important point is the login endpoint with your Tenant ID, instead of using the common endpoint.**

I'm using Azure Auth with Microsoft 365 accounts on Supabase via docker, and consuming it on a Redwoodjs project.

I remember struggling with this, and ended up with this configuration on my .env (used for docker compose):

```env
# Based on https://supabase.com/docs/guides/auth/social-login/auth-azure and https://github.com/supabase/gotrue#external-authentication-providers
# Your client ID
AZURE_CLIENT_ID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
AZURE_SECRET=*********************************
AZURE_REDIRECT_URI=http://localhost:8000/auth/v1/callback
# Your tenant ID
AZURE_URL=https://login.microsoftonline.com/yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy
```

Client side:

```js
const { isAuthenticated, logIn } = useAuth() //Redwood auth managment
const response = await logIn({
  authMethod: 'oauth',
  provider: 'azure',
  options: {
    scopes: 'email',
  },
})
```

Hope this helps!
